testcases/kernel/syscalls/readahead/readahead02.c

```diff
@@ -186,6 +188,23 @@ static void create_testfile(void)
 	free(tmp);
 }
 
+static void check_testfile(void)
+{
+	int fd;
+	int ret;
+	struct statfs sb;
+	fd = open(testfile, O_RDONLY);
+	ret = fstatfs(fd, &sb);
+	if (ret < 0)
+	{
+		tst_resm(TWARN, "Cannot detect %s locating file system", testfile);
+	}
+	if (sb.f_type == TMPFS_MAGIC)
+	{
+		tst_brkm(TCONF, NULL, "%s is locating on tmpfs which does not support readahead", testfile);
+	}
+	close(fd);
+}
 
```

Comment:
We do already have tst_fs_type() function exactly for this, so you are supposed to only write:

```
if (tst_fs_type(testfile) == TST_TMPFS_MAGIC)
        ....
```